chaos/litmus/master.yaml

```diff
@@ -0,0 +1,66 @@
+---
+# mayaonline can parse this yaml to obtain tests specific to applications (based on tag) & populate test params,
+# fulfill pre-requisites etc, before deployment. The params are explicitly typed to prompt right values from user.
+# Test parameters can be of different kinds such as ENV, configmap etc., with their own properties.
+# '@userdata' is a placeholder which is either autofilled/userfilled. If empty, use defaults. The test has tags which
+# will help with categorization & application-mapping
+
+
+litmus_test_1:
+  name: pod_kill
+  test_tags: ['mysql', 'pumba']
+  location: https://raw.githubusercontent.com/openebs/litmus/master/apps/percona/tests/mysql_data_persistence/run_litmus_test.yaml
+  test_params:
+    - name: DB_SERVER_IP
+      kind: env
+      type: string 
+      value: '@userdata'
```

Comment:
This yaml has `value` field, this makes me wonder if we going to have any logic in litmus side which will parse this yaml?

Reply:
The logic to fill/replace the values in the litmus job itself, will need to be performed by mayaonline. These values (in the env etc.,) will be consumed by the litmus test playbooks

Reply:
Following up on @a4abhishek question on `@userdata`:

The default (probably need to change to defaultValue), contains the string that needs to be replaced by a value that is determined by mayaonline.  For all the fields, we are specifying as @userdata, so we could probably eliminate the field for value, and MO can do the following:
- For each test_param, replace the defaultValue in the test URL with the MO determined value.   

To aid in the above logic, we can specify in each param, the YAML location of the value in the test yaml, for example: 
```
- name: PROVIDER_STORAGE_CLASS
  kind: env
  type: string
  defaultValue: "openebs-standard"
  spec: "spec.containers[name=ansibletest].env[name=PROVIDER_STORAGE_CLASS].value"
```

Reply:
`spec` field will be a great for agent.

Reply:
do we still need `kind` if we have `spec` field in this yaml?

Reply:
@a4abhishek , good question. I presume it will be useful in cases where there is a test param outside of spec, such as say, configmap

Reply:
@ksatchit can we not have respective `spec` for configmap as well?

Reply:
configmap will ideally be a separate resource @a4abhishek - it *maybe* bundled with the litmus job, but won't be part of the spec of the litmus pod itself.